### DISCUSSION TITLE
TCP links questions

### DISCUSSION THREAD

**Discussion by ziluChou** [rabbitmq/rabbitmq-perf-test#635] [General]
### Is your feature request related to a problem? Please describe.

When using perf-test, such as "java -jar perf-test.jar --queue-pattern 'perf-test-%d'  --queue-pattern-from 1 --queue-pattern-to 10 --producers 100 --consumers 100", will 100 TCP links be generated for the produces and 100 for the consumers?

### Describe the solution you'd like

When using perf-test, such as "java -jar perf-test.jar --queue-pattern 'perf-test-%d'  --queue-pattern-from 1 --queue-pattern-to 10 --producers 100 --consumers 100", will 100 TCP links be generated for the produces and 100 for the consumers?

**Comment by ziluChou** [comment]
Or AMQP links will multiplexing TCP links , for example, 100 amqp links ==> 10or20 TCP links？Thanks for your time.

**Comment by michaelklishin** [comment]
Using issues instead of GitHub discussions for questions is just rude.

One AMQP 0-9-1, one AMQP 1.0, one MQTT (any version), one STOMP connection, one MQTT-over-WebSockets connection, one STOMP-over-WebSockets connection [will use one TCP connection](https://rabbitmq.com/connections.html).

RabbitMQ stream protocol clients will [open parallel connections to all cluster nodes](https://blog.rabbitmq.com/posts/2021/07/connecting-to-streams/) unless explicitly configured otherwise.
